app/uk/gov/hmrc/exports/controllers/SubmissionController.scala

```diff
@@ -40,14 +41,23 @@ class SubmissionController @Inject() (
     extends RESTController(cc) with JSONResponses {
 
   def create(declarationId: String): Action[AnyContent] = authenticator.authorisedAction(parse.default) { implicit request =>
-    declarationService.markCompleted(request.eori, declarationId, declarationId).flatMap {
+    def submitDeclaration(declaration: ExportsDeclaration): Future[Submission] =
+      try
+        submissionService.submit(declaration)
+      catch {
+        case throwable: Throwable =>
+          logger.info(s"Declaration [${declaration.id}]: Submission failed")
+          declarationService.revertStatusToDraft(declaration, throwable)
+      }
```

Comment:
I think this reverting on exception should be the responsibility of the SubmissionService class rather than this controller.

Imagine if we wanted to call the `submit` method from another controller as well, then that call would be missing this protection.